src/ParallelDataCollector.cpp

```diff
@@ -475,14 +475,25 @@ namespace splash
         DCParallelGroup group;
         if (dataName)
         {
+            /** create `obj_name` as a group if not exists
+             *
+             * if the `obj_name` with the given `group_path` not exists than
+             *   - it is a group because datasets needs to be created before
+             *     attributes were added
+             *   - the group is created
+             */
+            std::string pathAndName(group_path + "/" + obj_name);
+            if(!DCParallelGroup::exists(handles.get(id), pathAndName))
```

Comment:
I don't get what this check does if `/path/to/something/<attribute>` is a data set. still working on existing data sets on appends attributes to?